apps/preview/next/pages/showcases/OrderSummary/OrderSummary.tsx

```diff
@@ -0,0 +1,93 @@
+import { ShowcasePageLayout } from '../../showcases';
+// #region source
+import { useState } from 'react';
+import { SfButton, SfInput, SfLink } from '@storefront-ui/react';
+
+const orderDetails = {
+  items: 3,
+  originalPrice: 7824.97,
+  savings: -787.0,
+  delivery: 0.0,
+  tax: 457.47,
+};
+
+export default function OrderSummary() {
+  const [inputValue, setInputValue] = useState('');
+  const [promoCode, setPromoCode] = useState(0);
+
+  const formatPrice = (price: number) =>
+    new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(price);
+
+  const itemsSubtotal = () =>
+    orderDetails.originalPrice + orderDetails.savings + orderDetails.delivery + orderDetails.tax;
+
+  const totalPrice = () => itemsSubtotal() + promoCode;
+
+  const checkPromoCode = () =>
+    inputValue.toUpperCase() === 'VSF2020' ? setPromoCode(-100) : console.log('Your promo code is not valid');
```

Comment:
```suggestion
    inputValue.toUpperCase() === 'VSF2020' ? setPromoCode(-100) : alert('Your promo code is not valid');
```
im not sure if displaying native alert would be better option now instead of hidden console.log, what do you think?

Reply:
You're right :) it will be clear that code is not valid and something's happened. I'll change it

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2579/commits/0857b9464bd93c263ac286d9636be4fc976cf218